datatypes/src/collections/feature_collection.rs

```diff
@@ -558,6 +570,71 @@ where
         ))
     }
 
+    /// Filter a column by one or more ranges
+    pub fn column_range_filter<R>(&self, column: &str, ranges: &[R]) -> Result<Self>
+    where
+        R: RangeBounds<FeatureDataValue>,
+    {
+        let column_type = self.types.get(column);
+        ensure!(
+            column_type.is_some(),
+            error::ColumnDoesNotExist {
+                name: column.to_string()
+            }
+        );
+
+        let column = self
+            .table
+            .column_by_name(column)
+            .expect("checked by ensure");
```

Comment:
ensure only checked if the column exists in types map

Reply:
Yes, but that is the assumption of the whole data structure.

Reply:
so it's ok that the whole process crashes if this assumption is violated?

Reply:
Yes, this assumption is used throughout the collection.